appium-dotnet-driver/Appium/ScreenRecording/BaseScreenRecordingOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using OpenQA.Selenium.Appium.Interfaces;
+
+namespace OpenQA.Selenium.Appium.ScreenRecording
+{
+    public abstract class BaseScreenRecordingOptions<T> : IScreenRecordingOptions where T : BaseScreenRecordingOptions<T>
+    {
+        protected Dictionary<string, object> Parameters;
+
+        protected BaseScreenRecordingOptions()
+        {
+            Parameters = new Dictionary<string, object>();
+        }
+
+        /// <summary>
+        /// Upload options set for the recorded screen capture.
+        /// </summary>
+        /// <param name="uploadOptions">Upload options</param>
+        /// <returns></returns>
+        public T WithUploadOptions(ScreenRecordingUploadOptions uploadOptions)
+        {
+            foreach (var parameter in uploadOptions.GetParameters())
```

Comment:
I'm not an expert in .net, but I believe there should be a shortcut for such basic operation like dictionary cloning

```
var newDictionary = oldDictionary.ToDictionary(entry => entry.Key, entry => entry.Value);
```

Reply:
In this example, a new dictionary is created, but I need to add some elements to the existing dictionary.